src/Toolkit/Collection.php

```diff
@@ -908,7 +908,13 @@ public function sortBy()
 
                     // make sure that we return something sortable
                     // but don't convert other scalars (especially numbers) to strings!
-                    $values[$key] = is_scalar($value) === true ? $value : (string)$value;
+                    $fieldValue = is_scalar($value) === true ? $value : (string)$value;
```

Comment:
In your commit description you write:

> This is for stable work. For ex: in files collection array values are empty, but pages collection array keys and values are filled.

To be honest I don't really understand why this change is needed. Is this related to the "Nesting level too deep - recursive dependency?" error or to a different issue? If so, please send me some details on that issue and why this change solves it.

Reply:
https://github.com/getkirby/kirby/pull/2020#issuecomment-531459943